### DISCUSSION TITLE
BU2D: horizontalGeometryEstimatedAccuracy (nilReason/uom)

### DISCUSSION THREAD

**Discussion by Dnoack49** [INSPIRE-MIF/helpdesk-validator#996] [Q&A] [answered]
Dear all,

1.) a description of the steps to reproduce your issue

We have validated the following Buildings-Dataset:
https://inspire.brandenburg.de/services/bu_alkis_wfs?service=WFS&version=2.0.0&request=GetFeature&Count=10&typenames=bu-core2d:Building

and receive errors, that `Attribute 'uom' must appear on element 'bu-base:horizontalGeometryEstimatedAccuracy'`

2.) used browser and version: `Firefox Browser 108.0.1 (64-Bit)`

3.) a screenshot illustrating the problem
![grafik](https://user-images.githubusercontent.com/8046960/214257105-2dcf9f93-e036-40ce-8aa6-74193d280324.png)

4.) the Test Report
[Test run on 09 54 - 24.01.2023 with test suite Annex III - Buildings (BU).zip](https://github.com/INSPIRE-MIF/helpdesk-validator/files/10488139/Test.run.on.09.54.-.24.01.2023.with.test.suite.Annex.III.-.Buildings.BU.zip)

5.) Name of the Test that failed :
The error appears in two conformance classes:

- CC: INSPIRE GML application schemas, General requirements // Validation against declared schema// gmlas.b.2: validate XML documents against declared schema
- CC: GML application schemas, Buildings // Validation against INSPIRE official schema//bu-gml.b.1: validate XML documents against official schema

According to the [UML-Model](https://inspire.ec.europa.eu/data-model/approved/r4618-ir/html/index.htm?goto=2:3:2:2:7911) the property **horizontalGeometryEstimatedAccuracy** can be voidable which it is in our dataset:

![grafik](https://user-images.githubusercontent.com/8046960/214259354-8d136882-485a-4c78-b187-0a53086bd885.png)

The question we have now: is it correct, that the **uom**-Property is expected for an attribute that has a nilReason-value? I f a value cannot be provided, why should a unit be given?
![grafik](https://user-images.githubusercontent.com/8046960/214259390-80db61f2-ab57-4de6-913c-4a9d70a5d780.png)

Kind regards

**Comment by fabiovinci** [comment, marked as answer]
Dear @Dnoack49,

good question!
It seems strange to provide a "uom" for a void element, but this "requirement" comes from the definition of the "MeasureType" data type defined in the BasicType.xsd of the GML specification. Since the "uom" is an attribute of the element it seems it is not possible to make it mandatory in another way.

![image](https://user-images.githubusercontent.com/22212406/214273214-d51262a1-ad6c-4246-a63a-fe85a2f50f27.png)

For the moment, you can by-pass the error by providing a dummy value, like: 

![image](https://user-images.githubusercontent.com/22212406/214289139-cd32d67c-21a5-4bc3-b022-f743a14e3bf3.png)
